File: Day2/calculator.php
<?php

function perform_operation($operator, $num1, $num2) {
    switch ($operator) {
        case "+":
            return $num1 + $num2;
        case "-":
            return $num1 - $num2;
        case "*":
            return $num1 * $num2;
        case "/":
            if ($num2 == 0) {
                return "Error: Division by zero";
            }
            return $num1 / $num2;
        default:
            return "Error: Invalid operator";
    }
}

while (true) {
    echo "Enter the first number: ";
    $num1 = trim(fgets(STDIN));
    if (!is_numeric($num1)) {
        echo "Invalid input. Please enter a number.\n";
        continue;
    }

    echo "Enter an operator (+, -, *, /): ";
    $operator = trim(fgets(STDIN));

    echo "Enter the second number: ";
    $num2 = trim(fgets(STDIN));
    if (!is_numeric($num2)) {
        echo "Invalid input. Please enter a number.\n";
        continue;
    }

    $result = perform_operation($operator, $num1, $num2);
    echo "Result: $result\n";

    echo "Would you like to perform another operation? (y/n): ";
    $continue = trim(fgets(STDIN));
    if ($continue !== 'y') {
        break;
    }
}
?>

File: Day2/calculator.py
def perform_operation(op, num1, num2):
    if op == "+":
        return num1 + num2
    elif op == "-":
        return num1 - num2
    elif op == "*":
        return num1 * num2
    elif op == "/":
        if num2 == 0:
            return "Error: Division by zero"
        else:
            return num1 / num2
    else:
        return "Error: Invalid operator"

def get_float_input(prompt):
    try:
        return float(input(prompt))
    except ValueError:
        return "Error: Invalid input"

def main():
    while True:
        num1 = get_float_input("Enter a number: ")
        op = input("Enter an operator (+,-,*,/): ")
        num2 = get_float_input("Enter another number: ")

        if isinstance(num1, str) or isinstance(num2, str):
            print("Error: Invalid input. Please enter numbers.")
            continue

        result = perform_operation(op, num1, num2)

        print(f"Result: {result}")

        another = input("Would you like to perform another operation? (y/n): ")
        if another.lower() != 'y':
            break

if __name__ == "__main__":
    main()
